lib/dynflow/dispatcher/client_dispatcher.rb

```diff
@@ -137,7 +137,14 @@ def dispatch_request(request, client_world_id, request_id)
                                AnyExecutor
                              end),
                             (on ~Event do |event|
-                               find_executor(event.execution_plan_id)
+                               id = find_executor(event.execution_plan_id)
+                               if id == Dispatcher::UnknownWorld && event.optional
+                                 message = "Could not find an executor for optional #{envelope}, discarding."
+                                 log(Logger::DEBUG, message)
+                                 respond(Envelope[request_id, client_world_id, id, request], Failed[message])
+                                 return
```

Comment:
Ugh... can't we have the notion of `optional` on the `Envelope`? Or at least pull it from here bellow?
```ruby
executor_id = nil
ignore_unknown = false
match request.....
( on ~ Event do |event|
  executor_id = find_executor...
  ignore_unknown = event.optional
end)
....
handle optinal here after match

Reply:
I went with something along the lines of your code snippet. Marking an entire envelope as optional felt wrong, it would mean there could be things which we always want to deliver in optional envelopes.